dts/arm/st/u0/stm32u073X8.dtsi

```diff
@@ -0,0 +1,30 @@
+
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <mem.h>
+#include <st/u0/stm32u073.dtsi>
+
+/ {
+	sram1: memory@20000000 {
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(32)>;
+		zephyr,memory-region = "SRAM1";
+	};
+
+	sram2: memory@20008000 {
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x20008000 DT_SIZE_K(8)>;
+		zephyr,memory-region = "SRAM2";
+	};
```

Comment:
Nitpick (non blocking), move the following directly  in stm32u073.dtsi:
```
	sram1: memory@20000000 {
		compatible = "zephyr,memory-region", "mmio-sram";
		zephyr,memory-region = "SRAM1";
	};

	sram2: memory@20008000 {
		compatible = "zephyr,memory-region", "mmio-sram";
		zephyr,memory-region = "SRAM2";
	};
```
and keep the following in this file:
```suggestion
	sram1: memory@20000000 {
		reg = <0x20000000 DT_SIZE_K(32)>;
	};

	sram2: memory@20008000 {
		reg = <0x20008000 DT_SIZE_K(8)>;
	};
```